### PR TITLE
Feature 8 tasks

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/vp-kuljetus-delivery-info-service-api/issues/8
For now Site/Freight removal triggers deleting the connected tasks, will be changed when the general deletion logic is modified. 
Site/Route/Freight fields are updatable for now. 

Adds env variable to check the routes: 
**quarkus.rest-client."fi.metatavu.vp.workplanning.spec.RoutesApi".url**

Depends on https://github.com/Metatavu/vp-kuljetus-delivery-info-service-api/pull/4

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            first = first,
+            max = max
+        )
+        createOk(tasks.map { taskTranslator.translate(it) }, count)
+    }.asUni()
+
+    @WithTransaction
+    @RolesAllowed(MANAGER_ROLE)
+    override fun createTask(task: Task): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        println("Creating task ${task.type}")
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt:79]:
```suggestion
```

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        routeId: UUID?,
+        assignedToRoute: Boolean?,
+        freightId: UUID?,
+        customerSiteId: UUID?,
+        type: TaskType?,
+        first: Int?,
+        max: Int?
+    ): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        val freightFilter = freightId?.let {
+            val foundFreight = freightController.findFreight(it) ?: return@async createNotFound(createNotFoundMessage(FREIGHT, it))
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt:55]:
```suggestion
            freightController.findFreight(it) ?: return@async createNotFound(createNotFoundMessage(FREIGHT, it))
```

File: src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightsTestIT.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package fi.metatavu.vp.deliveryinfo.functional
+
+import io.quarkus.test.junit.QuarkusIntegrationTest
+
+/**
+ * Native tests for Freights API
+ */
+@QuarkusIntegrationTest
```
Comment (villejuutila) [inline src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightsTestIT.kt:8]: Add testprofile to native tests

**Review by villejuutila — APPROVED** [review]

**Comment by kaeline** [comment]
Cannot be reviewed since the Task specs are not ready

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt (hunk 1 above)
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt:79]: removed

File: src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightsTestIT.kt (hunk 3 above)
Comment (kaeline) [reply src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightsTestIT.kt]: so far no special test profile was needed, once it is required, i will add it to native tests as well

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt (hunk 2 above)
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt:55]: Done

**Pull request merged**